developer_manual/core/ocs-share-api.rst

```diff
@@ -255,12 +255,12 @@ shareType    int     The type of the share. This can be one of: 0 = user,
 shareWith    string  The user or group id with which the file should be shared.
 publicUpload boolean Whether to allow public upload to a public shared folder.
 password     string  The password to protect public link share with.
-permissions  int     The permissions to set on the share. 
-                     1 = read; 2 = update; 4 = create; 8 = delete;
-                     16 = share; 31 = all (default: 31, for public shares: 1)
-expireDate   string  An expire date for public link shares. 
-                     This argument expects a date string in the following 
-                     format ``'YYYY-MM-DD'``.
+permissions  int     The permissions to set on the share.
+                     1 = read (default for public shares); 2 = update;
+                     4 = create; 8 = delete; 16 = share.
```

Comment:
@settermjd it may be worth noticing that default for internal shares (not public shares ones) seems to be 19. And also that permissions for public shares seems to be broken for versions < 10.0.3: here permissions are always returned as 1 regardless of what is given.

Reply:
Thanks for pointing that out @pako81